utils_test.go

```diff
@@ -34,6 +34,18 @@ func pollItems(ch chan interface{}, d time.Duration) []interface{} {
 	}
 }
 
+func pollTimestampedItems(ch chan interface{}, d time.Duration) []*Timestamped {
```

Comment:
Mmh I do not think we should create a test function per type. Well, I rather see what's coming next with Go generics. We would have a single typed `pollItems` function. For the time being (even if I agree this is not perfect), let's have only this `pollItems` and cast each item.

Reply:
Yeah, sounds good. I was a bit afraid of being too verbose on the test, but it makes sense :)